File: jobs/popular_subreddits.rb
recurring_job('1d') do
  listing = JSON.parse(Net::HTTP.get(URI('https://www.reddit.com/subreddits/popular.json')))

  list = listing['data']['children'].first(20).map do |item|
    item['data']['display_name']
  end
  {top_subreddits: {list: list}}
end

File: jobs/popular_subreddits.js
var request = require("request")
const url = "https://www.reddit.com/subreddits/popular.json";

module.exports = function() {
  return new Promise(function(fulfill, reject) {
    request(url, function (error, response, body) {
      if (!error && response.statusCode == 200) {
        var json = JSON.parse(body);
        var subreddit_list = [];
        json['data']['children'].slice(0,19).forEach(function(item) {
          subreddit_list.push(item['data']['display_name']);
        });
        fulfill({top_subreddits: {list: subreddit_list}});
      } else {
        reject(error);
      }
    });
  });
};
